### ISSUE TITLE
Circular reference when ingecting validator or sec-context and tagging doctrine onFlush event

### ISSUE THREAD

**Issue by Fludimir** [FriendsOfSymfony/FOSUserBundle#1111]
With FOSUserBundle its unable to create any event listener that requires validator or security-context and subscribed on doctrine onFlush event - you'll get exception

```
Circular reference detected for service "doctrine.orm.default_entity_manager", path: "doctrine.orm.default_entity_manager -> doctrine.dbal.default_connection -> acme.demo.brokenlistener_validator -> validator -> fos_user.user_manager".
```

example services:

``` xml
<!-- circular reference error when enabling fosuserbundle -->
        <service id="acme.demo.brokenlistener_validator" class="Acme\DemoBundle\EventListener\BrokenListenerWithValidator">
            <tag name="doctrine.event_listener" event="onFlush" method="onFlush" />
            <argument type="service" id="validator" />
        </service>
<!-- circular reference error when enabling fosuserbundle -->
        <service id="acme.demo.brokenlistener_sc" class="Acme\DemoBundle\EventListener\BrokenListenerWithSecurityContext">
            <tag name="doctrine.event_listener" event="onFlush" method="onFlush" />
            <argument type="service" id="security.context" />
        </service>
<!-- circular reference error in symfony-standard without fos
        <service id="acme.demo.brokenlistener_em" class="Acme\DemoBundle\EventListener\BrokenListenerWithEntityManager">
            <tag name="doctrine.event_listener" event="onFlush" method="onFlush" />
            <argument type="service" id="doctrine.orm.entity_manager" />
        </service>
        -->
<!-- solution for previous error in symfony-standard -->
        <service id="acme.demo.workinglistener_registry" class="Acme\DemoBundle\EventListener\WorkingListenerWithRegistry">
            <tag name="doctrine.event_listener" event="onFlush" method="onFlush" />
            <argument type="service" id="doctrine" />
        </service>
```

Only Symfony Standard 2.2 + FOSUserBundle (either 1.3.1 or current master) required to reproduce. Full code without vendors https://docs.google.com/file/d/0B1Q55deL45peWHc1dlFfbmt5ZkE/edit?usp=sharing .

Same problem http://www.9nit.com/php/injecting-securitycontext-services-into-a-listener-class-in-symfony2-causes-circular-reference-error-39161.html

**Comment by XWB** [comment]
Not related to this bundle.

**Comment by stof** [comment]
And btw, this should be fixed by #1615 as we don't inject the user manager in the validator anymore (we only inject the canonical fields updater)
